shell/platform/darwin/ios/framework/Source/FlutterViewControllerTest.mm

```diff
@@ -183,6 +184,36 @@ - (void)testViewDidDisappearDoesPauseEngineWhenIsTheViewController {
   XCTAssertNil(weakViewController);
 }
 
+- (void)testSurfaceUpdatedWillReturnWhenNotTheCurrentViewController {
+  FlutterEngine* mockEngine = OCMPartialMock([[FlutterEngine alloc] init]);
+  [mockEngine createShell:@"" libraryURI:@"" initialRoute:nil];
+  FlutterViewController* viewControllerA = [[FlutterViewController alloc] initWithEngine:mockEngine
+                                                                                 nibName:nil
+                                                                                  bundle:nil];
+  mockEngine.viewController = nil;
+  FlutterViewController* viewControllerB = [[FlutterViewController alloc] initWithEngine:mockEngine
+                                                                                 nibName:nil
+                                                                                  bundle:nil];
+  mockEngine.viewController = viewControllerB;
+  FlutterViewController* viewControllerMockA = OCMPartialMock(viewControllerA);
+  [viewControllerA surfaceUpdated:YES];
+  OCMVerify(never(), [viewControllerMockA installFirstFrameCallback]);
```

Comment:
Is it possible to look at the engine's view controller instead of relying on installFirstFrameCallback?